myscript.js

```diff
@@ -0,0 +1,198 @@
+
+const myUrl =  "http://localhost:5050/getFibonacciResults";
+let fibonacciUrl = "http://localhost:5050/fibonacci/";
+
+let myArr = [];
+let myIndex = document.getElementById("index");
+let myY = document.getElementById('y');
+let myMsg = document.getElementById('myMsg');
+let container = document.getElementById('containerDiv');
+let myCheckBox = document.getElementById('saveCalc');
+let isCalculationSaved; 
+let aListDropdown = document.getElementById('aListDropdown');
+
+window.addEventListener('load', init);
+
+aListDropdown.addEventListener('click', (e)=> {
+    container.innerHTML = "";
+    if (e.target.innerText  === "Number Asc") {   
+        spinnerOn();  
+        updateMyArr("Number Asc");
+    }
+    else if (e.target.innerText  === "Number Desc") {
+        spinnerOn();  
+        updateMyArr("Number Desc");
+    }
+    else if (e.target.innerText  === "Date Asc") {
+        spinnerOn();  
+        updateMyArr("Date Asc");
+    }
+    else if(e.target.innerText  === "Date Desc") {
+        spinnerOn();  
+        updateMyArr("Date Desc");
+    }
+});
+
+
+function spinnerOn() {
+    spinner.classList.add("spinner-border");
+    setTimeout(()=>spinner.classList.remove("spinner-border"),2000);
+}
+
+async function init(){
+    const response = await fetch(myUrl);
+    const data = await response.json();
+    myArr = data.results;
+    BuildResults();
```

Comment:
Instead of having a global array, you should call your BuildResults function with a parameter, which is an array: 
`BuildResults(sortedArray)
`

This will also solve the global array issue